transformers/modeling_seq2seq.py

```diff
@@ -155,22 +171,29 @@ def forward(self, encoder_input_ids, decoder_input_ids, **kwargs):
         """
         # Separate the encoder- and decoder- specific kwargs. A kwarg is
         # decoder-specific it the key starts with `decoder_`
-        kwargs_decoder = {}
-        kwargs_encoder = kwargs
-        for key in kwargs_encoder.keys():
-            if key.startswith("decoder_"):
-                kwargs_decoder[key.replace("decoder_", "")] = kwargs_encoder.pop(key)
+        kwargs_encoder = {
```

Comment:
Maybe we should have the same three possibilities here as well:
- arguments without an `encoder_` or `decoder_` prefix should be sent to both (e.g. a `training` argument for TF 2.0).
- arguments with `encoder_` should be sent to the encoder only (after removing the `encoder_` prefix)
- arguments with `decoder_` should be sent to the decoder only (after removing the `decoder_` prefix).

That would make it more coherent, what do you think?

Reply:
We would have to add the `encoder_hidden_states` and `encoder_attention_mask` to the decoder input as well.

Reply:
:+1: looks a lot better